contrib/lua/premake5.lua

```diff
@@ -21,8 +21,8 @@ project "lua-lib"
 		"etc/*.c"
 	}
 
-	configuration "linux or bsd or hurd or aix"
+	filter "system:linux or system:bsd or system:hurd or system:aix"
```

Comment:
> Currently yes. I'm not opposed to supporting "not system:windows" in addition to the current syntax though.
I wonder though, shouldn't [this](https://github.com/premake/premake-core/blob/master/tests/base/test_criteria.lua#L69) fail in that case then? Either I read the code for "not" wrong or the test isn't correct...

> Quite possible—feel free to add it!
done, also added printf and verbosef, someone might need to fill those pages with updated info which I lack (like removing the old doc styles and specifying when they were added)